app/assets/javascripts/components/generic_object/generic-object-definition-component.js

```diff
@@ -220,9 +220,11 @@ function genericObjectDefinitionFormController(API, miqService, $q) {
   }
 
   function buildOptionObject(optionData, optionObject) {
-    _.forEach(optionData, function(item) {
-      optionObject.push({id: item[1], name: item[0]});
-    });
+    for (var key in optionData) {
+      if optionData.hasOwnProperty(key) {
```

Comment:
syntax error ;) - missing parentheses around the conditional

But.. maybe `Object.keys(optionData).forEach(function(item) { ...` would save you that `hasOwnProperty` check :).

Reply:
Fixed.

Reply:
Is the `if optionData.hasOwnProperty(key) {`  conditional necessary?

...since we know that the incoming data will always have a `key`

My suggested change would be-
```javascript
function buildOptionObject(optionData, optionObject) {
  _.forEach(optionData, function(_value, key) {
    optionObject.push({id: key, name: __(optionData[key])});
  });
}
```